lib/errors.c

```diff
@@ -58,25 +148,83 @@ int nterror_to_errno(uint32_t status) {
+        case SMB2_STATUS_NO_SUCH_FILE:
+        case SMB2_STATUS_NO_SUCH_DEVICE:
         case SMB2_STATUS_BAD_NETWORK_NAME:
         case SMB2_STATUS_OBJECT_NAME_NOT_FOUND:
+        case SMB2_STATUS_OBJECT_PATH_INVALID:
+        case SMB2_STATUS_OBJECT_PATH_NOT_FOUND:
+        case SMB2_STATUS_OBJECT_PATH_SYNTAX_BAD:
+        case SMB2_STATUS_DFS_EXIT_PATH_FOUND:
+        case SMB2_STATUS_REDIRECTOR_NOT_STARTED:
                 return ENOENT;
         case SMB2_STATUS_FILE_CLOSED:
+        case SMB2_STATUS_SMB_BAD_FID:
+        case SMB2_STATUS_INVALID_HANDLE:
+        case SMB2_STATUS_OBJECT_TYPE_MISMATCH:
+        case SMB2_STATUS_PORT_DISCONNECTED:
+        case SMB2_STATUS_INVALID_PORT_HANDLE:
+        case SMB2_STATUS_HANDLE_NOT_CLOSABLE:
                 return EBADF;
         case SMB2_STATUS_MORE_PROCESSING_REQUIRED:
                 return EAGAIN;
         case SMB2_STATUS_ACCESS_DENIED:
+        case SMB2_STATUS_NETWORK_ACCESS_DENIED:
                 return EACCES;
-        case SMB2_STATUS_PENDING:
-                return EAGAIN;
+        case SMB2_STATUS_INVALID_LOCK_SEQUENCE:
+        case SMB2_STATUS_INVALID_VIEW_SIZE:
+        case SMB2_STATUS_ALREADY_COMMITTED:
+        case SMB2_STATUS_PORT_CONNECTION_REFUSED:
+        case SMB2_STATUS_THREAD_IS_TERMINATING:
+        case SMB2_STATUS_DELETE_PENDING:
+        case SMB2_STATUS_PRIVILEGE_NOT_HELD:
+        case SMB2_STATUS_FILE_IS_A_DIRECTORY:
+        case SMB2_STATUS_FILE_RENAMED:
+        case SMB2_STATUS_PROCESS_IS_TERMINATING:
+        case SMB2_STATUS_DIRECTORY_NOT_EMPTY:
+        case SMB2_STATUS_CANNOT_DELETE:
+        case SMB2_STATUS_FILE_DELETED:
+                return EPERM;
         case SMB2_STATUS_NO_MORE_FILES:
                 return ENODATA;
         case SMB2_STATUS_LOGON_FAILURE:
                 return ECONNREFUSED;
         case SMB2_STATUS_NOT_A_DIRECTORY:
                 return ENOTDIR;
+        case SMB2_STATUS_NOT_IMPLEMENTED:
+        case SMB2_STATUS_INVALID_DEVICE_REQUEST:
+        case SMB2_STATUS_ILLEGAL_FUNCTION:
         case SMB2_STATUS_INVALID_PARAMETER:
                 return EINVAL;
+        case SMB2_STATUS_TOO_MANY_OPENED_FILES:
+                return EMFILE;
+        case SMB2_STATUS_SECTION_TOO_BIG:
+        case SMB2_STATUS_TOO_MANY_PAGING_FILES:
+        case SMB2_STATUS_INSUFF_SERVER_RESOURCES:
+                return ENOMEM;
+        case SMB2_STATUS_NOT_SAME_DEVICE:
+                return EXDEV;
+        case SMB2_STATUS_SHARING_VIOLATION:
+                return ETXTBSY;
+        case SMB2_STATUS_FILE_LOCK_CONFLICT:
+        case SMB2_STATUS_LOCK_NOT_GRANTED:
+                return EDEADLK;
+        case SMB2_STATUS_OBJECT_NAME_COLLISION:
+                return EEXIST;
+        case SMB2_STATUS_PIPE_DISCONNECTED:
+                return EPIPE;
+        case SMB2_STATUS_MEDIA_WRITE_PROTECTED:
+                return EROFS;
+        case SMB2_STATUS_NO_MEDIA_IN_DEVICE:
+                return ENOMEDIUM;
```

Comment:
This error code is not avaiable in Darwin. Would you mind replace it with `ENODEV` or any other posix-defined error code?